File: TechChallengeWeb/Models/Publicacao.cs
using System.ComponentModel.DataAnnotations;

namespace TechChallengeWeb.Models
{
    public class Publicacao
    {
        public Publicacao(string nome, Usuario usuario, string urlPerfil, Foto foto)
        {
            Nome = nome;
            Usuario = usuario;
            UrlPerfil = urlPerfil;
            Foto = foto;
            DataEnvio = DateTime.Now;
        }

        [Key]
        [Required]
        public int Id { get; set; }
        public string Nome { get; set; }
        public Usuario Usuario { get; set; }
        public string UrlPerfil { get; set; }
        public Foto Foto { get; set; }
        public DateTime DataEnvio { get; set; }
    }
}

File: TechChallenge.WebApp/ClientApp/src/app/models/Publicacao.ts
import { Foto } from "./Foto";

export class Publicacao {
    constructor(nome:string, usuario:string, urlPerfil:string, foto:Foto)
    {
        this.nome = nome;
        this.usuario = usuario;
        this.urlPerfil = urlPerfil;
        this.foto = foto;
    }
    nome:string;
    usuario:string;
    foto:Foto;
    urlPerfil:string;
    id:number
}

